src/hotspot/cpu/x86/x86.ad

```diff
@@ -7311,57 +7323,66 @@ instruct evcmpFD(kReg dst, vec src1, vec src2, immI8 cond) %{
   ins_pipe( pipe_slow );
 %}
 
-instruct vcmp(legVec dst, legVec src1, legVec src2, immI8 cond, rRegP scratch) %{
+instruct vcmp_pri(legVec dst, legVec src1, legVec src2, immI8 cond) %{
   predicate(n->bottom_type()->isa_vectmask() == NULL &&
             !is_unsigned_booltest_pred(n->in(2)->get_int()) &&
             Matcher::vector_length_in_bytes(n->in(1)->in(1)) >=  4 && // src1
             Matcher::vector_length_in_bytes(n->in(1)->in(1)) <= 32 && // src1
-            is_integral_type(Matcher::vector_element_basic_type(n->in(1)->in(1)))); // src1
+            is_integral_type(Matcher::vector_element_basic_type(n->in(1)->in(1))) &&
+            (n->in(2)->get_int() == BoolTest::eq ||
```

Comment:
It's tempting to add a method to check the third bit of a `BoolTest` value, which controls the sense of the result e.g. `eq(0)` and `ne(4)`, rather than three separate checks e.g. `is_negated` perhaps. That it turn may result in more clearer naming of the methods rather than using `_pri` and `_sec`, and the logic percolates down into `vpcmpCCW` via the `ComparisonPredicate` value and the use of the tmp register.

Reply:
Thanks a lot for the suggestion. The situation is quite unique in this case and I can't find anywhere else the need for the check of the third bit. So I just renamed the nodes here to be more descriptive. What do you think?

Reply:
That's reasonable.